tuna/miopen/fin_class.py

```diff
@@ -107,23 +135,39 @@ def set_db_tables(self):
     self.dbt = MIOpenDBTables(session_id=self.session_id,
                               config_type=self.config_type)
 
-  def base_job_query(self, session):
-    """select tables for job query"""
-    query = session.query(self.dbt.job_table, self.dbt.config_table)
-    return query
-
-  def compose_work_query(self, query):
+  def compose_work_objs(self, session, conds):
     """query for fin command and config"""
+    ret = []
     if self.fin_steps:
```

Comment:
None, 0, and [] are evaluated as False by python. This one is a test against [], which would be the default value of fin_steps. If the array is populated it would branch into the first case here.